code/datums/components/damage_aura.dm

```diff
@@ -0,0 +1,142 @@
+#define DAMAGE_EFFECT_COOLDOWN (1 SECONDS)
+
+/// Applies a status effect and deals damage to people in the area.
+/// Will deal more damage the more people are present.
+/datum/component/damage_aura
+	/// The range of which to damage
+	var/range
+
+	/// Whether or not you must be a visible object of the parent
+	var/requires_visibility = TRUE
+
+	/// Brute damage to damage over a second
+	var/brute_damage = 0
+
+	/// Burn damage to damage over a second
+	var/burn_damage = 0
+
+	/// Toxin damage to damage over a second
+	var/toxin_damage = 0
+
+	/// Suffocation damage to damage over a second
+	var/suffocation_damage = 0
+
+	/// Stamina damage to damage over a second
+	var/stamina_damage = 0
+
+	/// Amount of cloning damage to damage over a second
+	var/clone_damage = 0
+
+	/// Amount of blood to damage over a second
+	var/blood_damage = 0
+
+	/// Map of organ (such as ORGAN_SLOT_BRAIN) to damage damage over a second
+	var/list/organ_damage = null
+
+	/// Amount of damage to damage on simple mobs over a second
+	var/simple_damage = 0
+
+	/// Which factions are immune to the damage aura
+	var/list/immune_factions = null
+
+	/// Sets a special set of conditions for the owner
+	var/datum/weakref/current_owner = null
```

Comment:
I still don't really love the ambiguity between "owner" (of this component) and "owner" (abstractly of an arbitrary atom which this component could be attached to) but I also don't have a good suggestion for an alternate name.

Reply:
Yeah, maybe special_target I thought about now, but oh well